utils/utils.jsx

```diff
@@ -1857,13 +1868,58 @@ export function applyHotkeyMarkdown(e) {
     let newEnd = 0;
     if (hasItalicAndBold || (hasCurrentMarkdown && !isItalicFollowedByBold)) {
         // message already has the markdown; remove it
-        newValue = prefix.substring(0, prefix.length - delimiter.length) + selection + suffix.substring(delimiter.length);
-        newStart = selectionStart - delimiter.length;
-        newEnd = selectionEnd - delimiter.length;
+        newValue = prefix.substring(0, prefix.length - delimiterStart.length) + selection + suffix.substring(delimiterEnd.length);
+        newStart = selectionStart - delimiterStart.length;
+        newEnd = selectionEnd - delimiterStart.length;
+    } else if (isLink) {
+        if (value.length === 0) {
+            // no input; Add [|](url)
+            newValue = delimiterStart + delimiterEnd;
+            newStart = 1;
+            newEnd = 1;
+        } else if (selectionStart < selectionEnd) {
+            // there is something selected; put markdown around it and preserve selection
```

Comment:
@hmhealey do you think there is a less complex way to go about this?

Reply:
I think the complexity might be necessary for handling all of these different cases, but perhaps we could organize it slightly differently to break it up a bit better.

Instead of having one function handling all 3 of these hotkeys, maybe we could break it up into different ones for italics/bold and links? The link hotkey works differently enough that it would make sense to keep on its own. In particular, the `delimiter` naming doesn't make as much sense for links since they're multiple parts, so splitting it out would let us rename them to something like `linkStart` and `linkEnd`.

Reply:
Done. Split it into two methods -- one for bold/italic and another for link.